May news (#203)
ropensci/roweb3: content/blog/2021-05-21-ropensci-news-digest-may-2021/index.Rmd

You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

We have no less than three categories of (online :wink:) events coming up: social co-working events, a community call, a conference with contributions from four rOpenSci staff members. :sparkles:

### Community Calls

* Our next community call is about [**rOpenSci's R-universe Project**](/commcalls/may2021-r-universe/) and will take place on Tuesday, 25 May 2021 17:00 UTC. The R-universe platform is a new umbrella project under which rOpenSci experiments with new ideas for improving publication and discovery of research software packages in R. In this 1-hour community call, **Jeroen Ooms** will explain the basic steps of setting up your own universe, and getting started with publishing packages (including experimental software, development versions, [research compendia](https://ropensci.org/blog/2021/04/09/r-universe-articles/)) and articles on your [personal subdomain](https://jeroen.r-universe.dev). We are interested in hearing your thoughts and ideas on how this infrastructure can improve how researchers publish and discover R packages and related content.

> * Ressources from our past community call [**Set Up Your Package to Foster a Community**](/commcalls/apr2021-pkg-community/) were posted. rOpenSci puts ongoing effort into exploring and communicating how developers can best attract attention to their package (e.g. usage, citations, or feedback), or how to set up their repository to encourage the types of contributions they want. In this 1-hour community call, **Maëlle Salmon**, **Hugo Gruson**, and **Steffi LaZerte** shared tips and examples on how to do this! **Sina Rüeger** [summarized the call](/blog/2021/04/28/commcall-pkg-community/) with help from Mark Padgham.

Reviewer: ```suggestion
* Video and resources from our past community call [**Set Up Your Package to Foster a Community**](/commcalls/apr2021-pkg-community/) were posted. rOpenSci puts ongoing effort into exploring and communicating how developers can best attract attention to their package (e.g. usage, citations, or feedback), or how to set up their repository to encourage the types of contributions they want. In this 1-hour community call, **Maëlle Salmon**, **Hugo Gruson**, and **Steffi LaZerte** shared tips and examples on how to do this! **Sina Rüeger** [summarized the call](/blog/2021/04/28/commcall-pkg-community/) with help from Mark Padgham.
```

### Social Co-Working

We held our first two **Social Co-Working and Label-athon events** on April 29th and May 13th. 
These were truly **international events** with 14 participants from Canada, Germany, Switzerland, USA, the UK, France, Singapore, and Mexico. 
We had a blast **socializing (scavenger hunts!) and co-working on our personal R projects** including repository organization, creating/labelling issues, preliminary research, personal GitHub README files, and those packages you can never find time for. 
We had **quiet and noisy rooms** to work in, so everyone was able to find an environment that worked best for them.
Sound like fun? We have **two more coming up**! 

> - [Thursday May 27th 9am Pacific](/events/labelathon03/) (16:00 UTC) - [Registration encouraged!](https://form.jotform.com/211236537172249).

Reviewer: ```suggestion
- [Thursday May 27th 9am Pacific](/events/labelathon03/) (16:00 UTC) - [Registration encouraged!](https://form.jotform.com/211236537172249)
```

Reviewer: already a ! at the end :-).

> - [Thursday June 10th 5pm Pacific](/events/labelathon04/) (Fri June 11th 00:00 UTC) - [Registration encouraged!](https://form.jotform.com/211236201763243).

Reviewer: ```suggestion
- [Thursday June 10th 5pm Pacific](/events/labelathon04/) (Fri June 11th 00:00 UTC) - [Registration encouraged!](https://form.jotform.com/211236201763243)
```

useR! conferences are non-profit conferences organized by community volunteers for the community, supported by the R Foundation.
useR! 2021 will take place online.

Get excited for four contributions by rOpenSci staff members:

* **Jeroen Ooms**, rOpenSci Lead Infrastructure Engineer, will give a [**keynote talk**](https://user2021.r-project.org/program/keynotes/) about **building a modern and scalable package build infrastructure**, such as R-universe.

* **Maëlle Salmon**, Research Software Engineer with rOpenSci, will give a [tutorial](https://user2021.r-project.org/program/tutorials/), _**GET better at testing your R package!**_ on Wednesday, July the 7th at 9-12AM GMT+2. This tutorial is about advanced testing of R packages, with HTTP testing as a case study.

* **Stefanie Butland**, rOpenSci community manager, will present on **rOpenSci's Model for Managing a Federated Open Source Software Community**.

* **Mark Padgham**, rOpenSci Software Research Scientist, will also give a regular talk, about his original **autotest** package for automatic testing of packages.

Find out about more [events](/events).

### Software review in Spanish

> Beside events, another great piece of news: censo2017 was the first package to pass **rOpenSci software peer-review in Spanish**! :tada:

Reviewer: ```suggestion
Another great piece of news: censo2017 was the first package to pass **rOpenSci software peer-review in Spanish**! :tada:
```

The [censo2017](https://github.com/ropensci/software-review/issues/414) package by [Pachá (aka Mauricio Vargas Sepúlveda)](/author/pachá-aka-mauricio-vargas-sepúlveda/) provides access to the 2017 Chilean Census.
Editor [Melina Vidoni](/blog/2019/01/31/more_editors/) was in charge of this review, the reviewers were [Frans van Dunné](https://github.com/FvD) and [María Paula Caldas](https://www.mpaulacaldas.com/).

> Frans van Dunné most kindly translated our [review template to Spanish](https://devguide.ropensci.org/reviewtemplatees.html)!

Reviewer: ```suggestion
Frans van Dunné most kindly translated our [review template to Spanish](https://devguide.ropensci.org/reviewtemplatees.html)!

Thank you to people in the Latin American R community for discussion and encouragement that helped make this happen.
```

There's no open call for new maintainers at this point but you can refer to our [contributing guide](https://contributing.ropensci.org/) for finding ways to get involved!
As the maintainer of an rOpenSci package, feel free to contact us on Slack or email `info@ropensci.org` to get your call for maintainer featured in the next newsletter. 

## Package development corner

Some useful tips for R package developers. :eyes:

Sometimes you find yourself documenting internal functions with roxygen2 without really wanting everyone to get to see the docs.
Do you know the [**difference between the `@internal` and `@NoRd` tags**](https://community.rstudio.com/t/keywords-internal-vs-nord/35119/3)?
* Keyword [`@internal`](https://roxygen2.r-lib.org/articles/rd.html#indexing) means there will be a manual page for the function but it won't be in the documentation index. " `@keywords internal` is useful because it removes the function from the documentation index; it’s useful for functions aimed primarily at other developers, not typical users of the package."
* `@NoRd` means there won't even be a manual page so the roxygen2 documentation of that function only lives in the R script. 

> We at rOpenSci [recommend not having manual pages for internal functions](https://devguide.ropensci.org/building.html): " Add `#' @noRd` to internal functions". It is also recommended in the [tidyverse style guide](https://style.tidyverse.org/documentation.html#internal-functions).

Reviewer 2: Could that maybe be tweaked a bit to *strongly encourage* documentation of internal functions? The more it is encouraged the better off we'll all be once the possibilities of automated code analysis really start to be explored and opened up. Something like:
> We at rOpenSci strongly encourage documentation of internal functions, and advise doing so within code itself rather than as separate manual pages -- that is, using `@noRd` and not `@internal`.

Author: thank you @mpadge! :pray:
